File: cmd/builder/go/network.go
package main

import (
	timeseries "github.com/grafana/cog/generated/timeseries/panel"
	types "github.com/grafana/cog/generated/types/dashboard"
)

func networkReceivedTimeseries() *timeseries.Builder {
	return defaultTimeseries().
		Title("Network Received").
		Description("Network received (bits/s)").
		Min(0).
		Unit("bps").
		FillOpacity(0).
		Targets([]types.Target{
			basicPrometheusQuery(`rate(node_network_receive_bytes_total{job="integrations/raspberrypi-node", instance="$instance", device!="lo"}[$__rate_interval]) * 8`, "{{ device }}"),
		})
}

func networkTransmittedTimeseries() *timeseries.Builder {
	return defaultTimeseries().
		Title("Network Transmitted").
		Description("Network transmitted (bits/s)").
		Min(0).
		Unit("bps").
		FillOpacity(0).
		Targets([]types.Target{
			basicPrometheusQuery(`rate(node_network_transmit_bytes_total{job="integrations/raspberrypi-node", instance="$instance", device!="lo"}[$__rate_interval]) * 8`, "{{ device }}"),
		})
}

File: examples/python/raspberry/network.py
import grafana_foundation_sdk.models.dashboard as dashboard
from grafana_foundation_sdk.cog import builder as cogbuilder
from .common import default_timeseries, basic_prometheus_query


def network_received_timeseries() -> cogbuilder.Builder[dashboard.Panel]:
    return (
        default_timeseries()
        .title("Network Received")
        .description("Network received (bits/s)")
        .min(0)
        .unit("bps")
        .fill_opacity(0)
        .with_target(
            basic_prometheus_query('rate(node_network_receive_bytes_total{job="integrations/raspberrypi-node", instance="$instance", device!="lo"}[$__rate_interval]) * 8', "{{ device }}")
        )
    )


def network_transmitted_timeseries() -> cogbuilder.Builder[dashboard.Panel]:
    return (
        default_timeseries()
        .title("Network Transmitted")
        .description("Network transmitted (bits/s)")
        .min(0)
        .unit("bps")
        .fill_opacity(0)
        .with_target(
            basic_prometheus_query('rate(node_network_transmit_bytes_total{job="integrations/raspberrypi-node", instance="$instance", device!="lo"}[$__rate_interval]) * 8', "{{ device }}")
        )
    )

